src/conn/conn_layered.c

```diff
@@ -745,6 +730,7 @@ __wti_layered_table_manager_destroy(WT_SESSION_IMPL *session)
      * Spin until exclusive access is gained. If we got here from the startup path seeing an error,
      * the state might still be "starting" rather than "running".
      */
+    /* FIXME-WT-14963: Check whether we need atomics here. */
     while (!__wt_atomic_cas32(&manager->state, WT_LAYERED_TABLE_MANAGER_RUNNING,
```

Comment:
This is something I think we should resolve in this PR. If it's not atomic any more -- and I don't think it is -- we can remove the CAS and remove the `wt_shared` marking on the variable itself. It also lets us simplify the logic below where do a retry dance to put it in the `STOPPING` state.

Reply:
I would be happy to do this, but I am a bit worried why we think it should be fine to observe `WT_LAYERED_TABLE_MANAGER_STOPPING` from `__wt_layered_table_manager_remove_table`. Please see the following assertion: https://github.com/wiredtiger/wiredtiger/blob/25c3a5e7a1ec74902c18cba46db1eb14470e2543/src/conn/conn_layered.c#L681-L683

Reply:
Ok, I read the code a bit more and now I agree with you, done, thanks for pointing that out!